File: app/src/main/webapp/js/classes/stream-case.js
class StreamCase {
    static fromJson(json) {
        if (json == null) {
            return null;
        }

        const streamCase = new StreamCase();

        streamCase.setId(json.id);
        streamCase.setStreamPositionX(json.streamPositionX);
        streamCase.setStreamPositionY(json.streamPositionY);

        for (let i in json.streamParameters) {
            let streamParameter = StreamCaseParameter.fromJson(json.streamParameters[i]);
            streamCase.addStreamParameter(streamParameter);
        }

        for (let i in json.streamSignals) {
            let streamSignal = StreamCaseSignal.fromJson(json.streamSignals[i]);
            streamCase.addStreamSignal(streamSignal);
        }

        return streamCase;
    }

    static toJson(streamCase) {
        if (streamCase == null) {
            return null;
        }
        const streamCaseJson = {
            id:                 streamCase.getId(),
            streamPositionX:    streamCase.getStreamPositionX(),
            streamPositionY:    streamCase.getStreamPositionY()
        }

        addConvertedJsonArray(streamCaseJson, "streamParameters",   streamCase.getStreamParameters(),   StreamCaseParameter.toJson);
        addConvertedJsonArray(streamCaseJson, "streamSignals",      streamCase.getStreamSignals(),      StreamCaseSignal.toJson);
    }

    constructor() {
        this._id                = null;
        this._streamPositionX   = null;
        this._streamPositionY   = null;
        this._streamParameters  = [];
        this._streamSignals     = [];
    }

    setId(id) {
        this._id = id;
    }

    getId() {
        return this._id;
    }

    setStreamPositionX(streamPositionX) {
        this._streamPositionX = streamPositionX;
    }

    getStreamPositionX() {
        return this._streamPositionX;
    }

    setStreamPositionY(streamPositionY) {
        this._streamPositionY = streamPositionY;
    }

    getStreamPositionY() {
        return this._streamPositionY;
    }

    setStreamParameters(streamParameters) {
        this._streamParameters = streamParameters;
    }

    addStreamParameter(streamParameter) {
        this._streamParameter.push(streamParameter);
    }

    getStreamParameters() {
        return this._streamParameters;
    }

    setStreamSignals(streamSignals) {
        this._streamSignals = streamSignals;
    }

    addStreamSignal(streamSignal) {
        this._streamSignal.push(streamSignal);
    }

    getStreamSignals() {
        return this._streamSignals;
    }

}

File: app/src/main/java/com/softwareverde/tidyduck/most/StreamCase.java
package com.softwareverde.tidyduck.most;

import java.util.ArrayList;
import java.util.List;

public class StreamCase {
    private Long _id;
    private String _streamPositionX;
    private String _streamPositionY;
    private List<StreamCaseParameter> _streamCaseParameters = new ArrayList<>();
    private List<StreamCaseSignal> _streamCaseSignals = new ArrayList<>();

    public Long getId() {
        return _id;
    }

    public void setId(Long id) {
        _id = id;
    }

    public String getStreamPositionX() {
        return _streamPositionX;
    }

    public void setStreamPositionX(String streamPositionX) {
        _streamPositionX = streamPositionX;
    }

    public String getStreamPositionY() {
        return _streamPositionY;
    }

    public void setStreamPositionY(String streamPositionY) {
        _streamPositionY = streamPositionY;
    }

    public List<StreamCaseParameter> getStreamCaseParameters() {
        return new ArrayList<>(_streamCaseParameters);
    }

    public void addStreamCaseParameter(final StreamCaseParameter streamCaseParameter) {
        _streamCaseParameters.add(streamCaseParameter);
    }

    public void setStreamCaseParameters(final List<StreamCaseParameter> streamCaseParameters) {
        _streamCaseParameters = new ArrayList<>(streamCaseParameters);
    }

    public List<StreamCaseSignal> getStreamCaseSignals() {
        return new ArrayList<>(_streamCaseSignals);
    }

    public void addStreamCaseSignal(final StreamCaseSignal streamCaseSignal) {
        _streamCaseSignals.add(streamCaseSignal);
    }

    public void setStreamCaseSignals(final List<StreamCaseSignal> streamCaseSignals) {
        _streamCaseSignals = new ArrayList<>(streamCaseSignals);
    }
}
